etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiOrder.java

```diff
@@ -42,4 +50,170 @@ public String getPatientId() {
                 .findFirst()
                 .orElse("");
     }
+
+    @Override
+    public String getPlacerOrderNumber() {
+        return null;
+    }
+
+    @Override
+    public String getSendingApplicationDetails() {
+        return HapiHelper.resourcesInBundle(innerOrder, MessageHeader.class)
+                .filter(Objects::nonNull)
+                .findFirst()
+                .map(
+                        messageHeader -> {
+                            String namespaceId = null;
+                            String universalId = null;
+                            String universalIdType = null;
+                            String endpoint = messageHeader.getSource().getEndpoint();
+
+                            for (Extension extension : messageHeader.getSource().getExtension()) {
+                                if ("https://reportstream.cdc.gov/fhir/StructureDefinition/namespace-id"
+                                        .equals(extension.getUrl())) {
+                                    namespaceId =
+                                            Objects.toString(
+                                                    extension.getValue().primitiveValue(), "");
+                                } else if ("https://reportstream.cdc.gov/fhir/StructureDefinition/universal-id"
+                                        .equals(extension.getUrl())) {
+                                    universalId =
+                                            Objects.toString(
+                                                    extension.getValue().primitiveValue(), "");
+                                } else if ("https://reportstream.cdc.gov/fhir/StructureDefinition/universal-id-type"
+                                        .equals(extension.getUrl())) {
+                                    universalIdType =
+                                            Objects.toString(
+                                                    extension.getValue().primitiveValue(), "");
+                                }
+                            }
+
+                            return concatenateWithCaret(
+                                    namespaceId, universalId, universalIdType, endpoint);
```

Comment:
I'm not sure if the endpoint is needed here. I think we can go without it, thoughts?